PR: https://github.com/odoo/odoo/pull/33863

From: ed18095127b20da516fa5ada943b2dd6d5c84652
From: Julien Mougenot
Date: 2019-07-29 08:29:26

Breaking data model changes score: 3, change matches:
-    view_id = fields.Many2one('ir.ui.view', 'Document Template', required=True)
-    image = fields.Char(string="Preview image src")
-    pdf = fields.Char(string="Preview pdf src")

Total Changes: 19

[IMP] base: Configure document layout

The onboarding modal for setting up the few base fields of a company
has now been moved to a wizard
It is accessible from the general settings, but also in the onboarding
section of sale and account modules.

The following company settings are editable with that wizard:

- Set report **layout**:
The user can chose the overall look of the report. The current choices
are : *Standard* (default), *Background*, *Boxed* and *Clean*.

- Set company **logo**:
Changes the company logo.

- Set report **colors**:
The user can set the primary and secondary colors of the report through
a newly added widget allowing to pick a custom color.
When changing the **logo**, colors are automatically set to its most dominant
colors.
> A "Reset colors" button also triggers the color calculation.

- Set report **font**:
Changes the overall font of the report. Only Google Fonts are used
for enhanced compatibility.

- Company **tagline**, also called "header"
- **Footer**
- **Paper format**
- Report **preview**:
A mockup of a final report
Automatically updates when changing **layout**, **logo**, **colors** or **font**

Co-authored by: Julien Mougenot <jum@odoo.com>

closes odoo/odoo#33863

Signed-off-by: VincentSchippefilt <VincentSchippefilt@users.noreply.github.com>


Co-authored-by: Lucas Perais <lpe@odoo.com>

================================= pseudo patch: =================================

--- a/addons/web/models/__init__.py
+++ b/addons/web/models/__init__.py
@@ -4,4 +4,3 @@
 from . import ir_qweb
 from . import ir_http
 from . import models
-from . import report_layout

--- a/addons/web/models/report_layout.py
+++ b/None
@@ -1,18 +0,0 @@
-# -*- coding: ascii -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-import logging
-
-from odoo import fields, models
-
-_logger = logging.getLogger(__name__)
-
-
-class View(models.Model):
-
-    _name = "report.layout"
-    _description = 'Report Layout'
-
-    view_id = fields.Many2one('ir.ui.view', 'Document Template', required=True)
-    image = fields.Char(string="Preview image src")
-    pdf = fields.Char(string="Preview pdf src")
